### ISSUE TITLE
Syslog based plugins fail because tag length is too long

### ISSUE THREAD

**Issue by jasonkeene** [vmware/kube-fluentd-operator#11]
We've tried both papertrail and the remote_syslog plugins and both use the syslog_protocol gem that follows RFC 3164 which limits tag length to 32 characters. We were getting this stack trace in our fluentd container logs:

```
2018-06-01 16:48:36 +0000 [warn]: #0 failed to flush the buffer. retry_time=0 next_retry_seconds=2018-06-01 16:48:37 +0000 chunk="56d975adcff66f1a1e87fce8bdd3305a" error_class=ArgumentError error="Tag must not be longer than 32 characters"
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/syslog_protocol-0.9.2/lib/syslog_protocol/packet.rb:49:in `tag='
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluent-plugin-papertrail-0.2.4/lib/fluent/plugin/out_papertrail.rb:64:in `create_packet'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluent-plugin-papertrail-0.2.4/lib/fluent/plugin/out_papertrail.rb:36:in `block in write'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/event.rb:323:in `each'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/event.rb:323:in `block in each'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/plugin/buffer/memory_chunk.rb:80:in `open'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/plugin/buffer/memory_chunk.rb:80:in `open'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/event.rb:322:in `each'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluent-plugin-papertrail-0.2.4/lib/fluent/plugin/out_papertrail.rb:34:in `write'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/compat/output.rb:131:in `write'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/plugin/output.rb:1096:in `try_flush'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/plugin/output.rb:1329:in `flush_thread_run'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/plugin/output.rb:439:in `block (2 levels) in start'
  2018-06-01 16:48:36 +0000 [warn]: #0 /var/lib/gems/2.3.0/gems/fluentd-1.1.3/lib/fluent/plugin_helper/thread.rb:78:in `block in thread_create'
```

When we log out the tag it is:

`kube.demo.nginx-65899c769f-5zj6d.nginx`

which is indeed longer than 32 characters. This seems like a bug in the operator since it is creating these extra long tags. Perhaps they can be truncated? An alternative solution might be to support an output plugin for syslog that uses RFC 5424 and structured data for this info?

cc/ @wfernandes @ahevenor @st3v

**Comment by jvassev** [comment]
Hi @jasonkeene,

That's definitely unfortunate - the tag is limited by the spec. Even if you have a server that is more lax, the syslog ruby library used by the `remote_syslog` plugin validates this strictly.

So, the best solution is to strip the tag of the unimportant bits. I've implemented a fix in https://github.com/vmware/kube-fluentd-operator/commit/faf533d887eef5d28a68d5ecf7c8d9199c8ec925 and have successfully tested it against papertrail.

Can you try with the new image too. Just use these for Helm `--set image.repository=jvassev/kube-fluentd-operator --set image.tag=v1.2.0-10-gfaf533d`.

The only change you need to make is replace `remote_syslog` with `truncating_remote_syslog` in your `<match *>` directive.

**Comment by jvassev** [comment]
I am closing the issue since the fix made into the v1.3.0 release. You can now use the v1.3.0 tag for the image as well.
Feel free to reopen if there are still issues with remote syslog.
